src/receiveParser.py

```diff
@@ -26,8 +26,8 @@ def __init__(self):
 
     def parseReturnValue(self, src, dport, data):
         systemType = None
-        for node in SatelliteNodes:
-            if node[2] == src:
+        for node in SatelliteNodes:     # A: Hmm, there is scope of improvement to do this.
```

Comment:
I'm curious to hear what your suggested improved way of changing this approach would be. Maybe a dictionary separated into ground station nodes, and satellite nodes, consisting of associated satellites or ground station things ( [1::] elements in the tuples) ?

Reply:
Yup, a dictionary is a candidate. The key could just be the third numerical element. For instance, `nodes = {1: ("OBC", "EX2"), ...}`, and then this whole block of code is replaced by one line: `systemType = nodes[src][0]`.

And given this piece of code is going through both lists one-by-one, can we combine the two lists/dictionaries or is there is a good reason to keep them separate?